packages/vats/src/lib-board.js

```diff
@@ -1,22 +1,55 @@
 // @ts-check
+/**
+ * @file lib-board: share objects by way of plain-data ids
+ *
+ * @see prepareBoardClass()
+ */
 
-import { assert, details as X, q, Fail } from '@agoric/assert';
+import { assert, q, Fail } from '@agoric/assert';
 import { E, Far } from '@endo/far';
 import { makeMarshal } from '@endo/marshal';
-import { makeScalarMapStore } from '@agoric/store';
-import { crc6 } from './crc.js';
+import {
+  M,
+  makeScalarBigMapStore,
+  prepareExoClassKit,
+  provide,
+} from '@agoric/vat-data';
+import { makeEphemeraProvider } from '@agoric/zoe/src/contractSupport/durability.js';
 
-import './types.js';
+import { crc6 } from './crc.js';
 
 export const DEFAULT_CRC_DIGITS = 2;
 export const DEFAULT_PREFIX = 'board0';
 
+// #region Interface Guards
+const CapDataShape = { body: M.string(), slots: M.array() };
+const MarshalI = M.interface('Marshaller', {
+  serialize: M.call(M.any()).returns(CapDataShape),
+  unserialize: M.call(CapDataShape).returns(M.any()),
+});
+
+const IdShape = M.string();
+const ValShape = M.key();
+const BoardI = {
+  board: M.interface('Board', {
+    has: M.call(ValShape).returns(M.boolean()),
+    getValue: M.call(IdShape).returns(ValShape),
+    lookup: M.callWhen().rest(M.arrayOf(IdShape)).returns(ValShape),
```

Comment:
taking another look at https://github.com/Agoric/agoric-sdk/wiki/Interfaces , I suppose this should be:

```suggestion
    lookup: M.call().rest(M.arrayOf(IdShape)).returns(M.promise(ValShape)),
```